src/main/java/codeu/model/store/persistence/PersistentDataStore.java

```diff
@@ -66,7 +67,13 @@ public PersistentDataStore() {
         String userName = (String) entity.getProperty("username");
         String password = (String) entity.getProperty("password");
         Instant creationTime = Instant.parse((String) entity.getProperty("creation_time"));
-        User user = new User(uuid, userName, password, creationTime);
+        int messageCount = 0;
+        if (entity.hasProperty("messageCount")){
+          messageCount = (int) (long) entity.getProperty("messageCount");
```

Comment:
What's the double casting for?

Reply:
when i only do (int) it gives me an error saying cannot cast long to int, but when i do (int) (long) it works. I don't rly get why too.